internal/db/queries.go

```diff
@@ -36,12 +36,42 @@ func CreateCommitEntry(db *gorm.DB, c []models.Commits) *gorm.DB {
 	return db
 }
 
-func GetServicesAll(db *gorm.DB) (*gorm.DB, []models.Services) {
+// GetAllByServiceName returns all commits for a given service
+// TODO: this should include deploys once we have support for that
+func GetAllByServiceName(db *gorm.DB, name string) (*gorm.DB, models.Services) {
+	callDurationTimer := prometheus.NewTimer(metrics.SqlGetAllByServiceName)
+	defer callDurationTimer.ObserveDuration()
+	var services models.Services
+	l.Log.Debugf("Query name: %s", name)
+	db.Table("services").Select("*").Where("name = ?", name).First(&services)
+	result := db.Table("commits").Select("*").Joins("JOIN services ON commits.service_id = services.id").Where("services.name = ?", name).Order("Timestamp desc").Find(&services.Commits)
+  services.Deploys = []models.Deploys{}
+	return result, services
+}
+
+func GetLatest(db *gorm.DB, service models.Services) (*gorm.DB, models.Services) {
+	callDurationTimer := prometheus.NewTimer(metrics.SqlGetAllByServiceName)
+	defer callDurationTimer.ObserveDuration()
+	l.Log.Debugf("Query name: %s", service.Name)
+	result := db.Table("commits").Select("*").Joins("JOIN services ON commits.service_id = services.id").Where("services.name = ?", service.Name).Order("Timestamp desc").Limit(1).Find(&service.Commits)
+	return result, service
+}
+
+func GetServicesAll(db *gorm.DB) (*gorm.DB, []models.Services, []models.Services) {
 	callDurationTimer := prometheus.NewTimer(metrics.SqlGetServicesAll)
 	defer callDurationTimer.ObserveDuration()
 	var services []models.Services
 	result := db.Find(&services)
-	return result, services
+
+	var latest_commits []models.Services
+	for i := 0; i < len(services); i++ {
+
```

Comment:
<nitpick />

It might be better to change the variable name to something a little more descriptive/accurate like 
`servicesWithCommits`.

Also, golang naming convention suggests to use camelCase instead of snake_case to name variables.
So, `latest_commits` would be `latestCommits`.